### PR TITLE
.github: Add workflow to build conda packages

### PR DESCRIPTION
**Pull request by seemethere**
Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

### CODE REVIEW

**Comment by seemethere** [comment]
Successful workflow at 7185f8b

File: .github/scripts/generate_binary_build_matrix.py [hunk 1]
```diff
@@ -54,33 +58,57 @@
...
+        arches += cuda_versions
+    if include_rocm:
+        arches += rocm_versions
     matrix = []
     for item in itertools.product(python_versions, arches):
         python_version, arch_version = item
         # Not my favorite code here
         gpu_arch_type = "cuda"
-        if "rocm" in CONTAINER_IMAGES[arch_version]:
+        if "rocm" in WHEEL_CONTAINER_IMAGES[arch_version]:
```
Comment (walterddr) [inline .github/scripts/generate_binary_build_matrix.py:82]: this seems to assume the same arch_version exists on both conda and wheel image dict keyset. yes?

**Review by walterddr — APPROVED** [review]
looks good overall. one question regarding wheel conda arch_version support policy. Seems we are making assumption that wheel image support version is a superset of conda. If this is the right assumption we are good to go

File: .github/scripts/generate_binary_build_matrix.py (hunk 1 above)
Comment (seemethere) [reply .github/scripts/generate_binary_build_matrix.py:82]: Unfortunately not necessarily, conda does not have a rocm equivalent just yet

File: .github/scripts/generate_binary_build_matrix.py [hunk 2]
```diff
@@ -45,6 +39,16 @@
     "cpu": "pytorch/manylinux-cpu"
 }
 
+CONDA_CONTAINER_IMAGES = {
```
Comment (malfet) [inline .github/scripts/generate_binary_build_matrix.py:42]: Please consider type-annotating your code
```suggestion
CONDA_CONTAINER_IMAGES : Dict[str, str]= {
```

File: .github/scripts/generate_binary_build_matrix.py [hunk 3]
```diff
@@ -54,33 +58,57 @@
 
 
 def is_pull_request():
-    return os.environ.get("GITHUB_HEAD_REF")
+    return False
+    # return os.environ.get("GITHUB_HEAD_REF")
 
-def generate_matrix():
+def generate_matrix(include_cuda=True, include_rocm=False):
```
Comment (malfet) [inline .github/scripts/generate_binary_build_matrix.py:64]:
```suggestion
def generate_matrix(include_cuda: bool = True, include_rocm:bool = False) -> str:
```

**Review by malfet — APPROVED** [review]

File: .github/scripts/generate_binary_build_matrix.py (hunk 1 above)
Comment (walterddr) [reply .github/scripts/generate_binary_build_matrix.py:82]: oops. my mistake. I meant wheel container dict keyset will be a superset of conda container dict keyset

**Pull request closed without merging**